sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sqs/MessageCoder.java

```diff
@@ -1,50 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.beam.sdk.io.aws2.sqs;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.io.Serializable;
-import org.apache.beam.sdk.coders.AtomicCoder;
-import org.apache.beam.sdk.coders.StringUtf8Coder;
-import software.amazon.awssdk.services.sqs.model.Message;
-
-/** Custom Coder for handling SendMessageRequest for using in Write. */
-public class MessageCoder extends AtomicCoder<Message> implements Serializable {
```

Comment:
This is public and not `@Internal` so technically we shouldn't just remove it outright. That being said this and `SendMessageRequestCoder`, `MessageCoder` are pretty clearly internal coders for use in their respective IOs, so it's probably fine.

I'll leave it up to you - if you want you could mark these coders deprecated and schedule them for removal in a later release.

Reply:
You're absolutely right pointing this out 👍 Considering `MessageCoder.of()` / `SendMessageRequestCoder.of()` are both package private I think direct removal is justifiable in this case.